### ISSUE TITLE
--rerun-failed executes all tests

### ISSUE THREAD

**Issue by damoon** [vmware-tanzu/sonobuoy#1387]
**What steps did you take and what happened:**
./sonobuoy run --wait --plugin-env=e2e.E2E_PARALLEL=y
one test failed
./sonobuoy retrieve
./sonobuoy e2e 202108130924_sonobuoy_165ec6ae-f8e6-4f9d-9361-5635dc1fcefc.tar.gz --rerun-failed --plugin-env=e2e.E2E_PARALLEL=y --wait
all tests executed again

**What did you expect to happen:**
only one test would be executed

**Anything else you would like to add:**
looks like https://github.com/vmware-tanzu/sonobuoy/issues/797 resurfaced

**Environment:**

- Sonobuoy version: v0.53.1
- Kubernetes version: v1.21.3
- Kubernetes installer & version: kubeadm v1.21.3
- Cloud provider or hardware configuration: bare metal
- OS (e.g. from `/etc/os-release`): Flatcar Container Linux by Kinvolk 2905.2.1 (Oklo)
- Sonobuoy tarball (which contains * below)

**Comment by johnSchnake** [comment]
Working on repro/identifying the issue right now. First try and I found that when I did quick mode (1 test) and it passed, when doing rerun-failed on the tarball it ended up just running the whole suite as well. When I looked at the logs it looked like the focus was the whole conformance suite explicitly too.

**Comment by johnSchnake** [comment]
Ah, I think I see the issue though I'll have to confirm it and figure out the best way to test it.

It sets the PluginEnvOverrides value but I think that probably is having some conflict with the recent changes to how the focus/skip were being handled now (e.g. a transform on the plugin def before being run). TBD

**Comment by johnSchnake** [comment]
Looking at the e2e command in general I am leaning towards:
 - marking it as deprecated since the 'show' type functionality is duplicated by the newer 'results' command and gen/run commands already exist.
 - Introduce a new flag/command to simply spit out a regexp from the tarball's failed tests.

E.g.:

```
sonobuoy gen/run --failed-regexp-from-tar output.tar.gz ...
```

One new flag (which needs some work on the naming) to actually _increase_ functionality since now it would work with `gen` as well and we get rid of an entire command.

**Comment by johnSchnake** [comment]
So far I've been unable to repro, even with v0.53.1 of Sonobuoy.

**Comment by johnSchnake** [comment]
Just merged #1395 so in the next release `e2e` will be gone and `gen/run` will have a `--rerun-failed <tarball>` flag that accomplishes the same thing as `sonobuoy e2e <tarball> --rerun-failed`.
